src/pages/Admin/Venue/DetailsForm.tsx

```diff
@@ -823,8 +867,79 @@ const DetailsFormLeft: React.FC<DetailsFormLeftProps> = ({
     </div>
   );
 
+  const { sovereignVenueId } = useSovereignVenueId({ venueId });
+
+  const { currentVenue: venue } = useConnectCurrentVenueNG(sovereignVenueId);
+
+  const [userStatuses, setUserStatuses] = useState<UserStatus[]>([]);
+
+  const {
+    isShown: hasUserStatuses,
+    show: showUserStatuses,
+    toggle: toggleUserStatus,
+  } = useShowHide(venue?.showUserStatus);
+
+  // Because this is not using the useForm validation. The use effect needs to manually open the dropdown with user statuses.
+  useEffect(() => {
+    if (!venue) return;
+    const venueUserStatuses = venue?.userStatuses ?? [];
+
+    setUserStatuses(venueUserStatuses);
+
+    if (venue.showUserStatus) {
+      showUserStatuses();
+    }
+  }, [showUserStatuses, venue]);
+
+  const removeUserStatus = (index: number) => {
+    const statuses = [...userStatuses];
+    statuses.splice(index, 1);
+    setUserStatuses(statuses);
+  };
+
+  const addUserStatus = () =>
+    setUserStatuses([...userStatuses, { status: "", color: "#4BCC4B" }]);
+
+  const updateStatusColor = (color: string, index: number) => {
+    const statuses = [...userStatuses];
+    statuses[index] = { status: statuses[index].status, color: color };
```

Comment:
```suggestion
    statuses[index] = { color, status: statuses[index].status };
```

Reply:
Changed in: https://github.com/sparkletown/sparkle/pull/1532/commits/0c1f9f07a5696848014d78ad2604de3c1ed2d194